src/language/templates/visitors/ckparent_visitor.hpp

```diff
@@ -0,0 +1,58 @@
+/*************************************************************************
+ * Copyright (C) 2018-2019 Blue Brain Project
+ *
+ * This file is part of NMODL distributed under the terms of the GNU
+ * Lesser General Public License. See top-level LICENSE file for details.
+ *************************************************************************/
+
+#pragma once
+
+/**
+ *
+ * \dir
+ * \brief Auto generated visitors
+ *
+ * \file
+ * \brief \copybrief nmodl::visitor::CkParentVisitor
+ */
+
+#include "visitors/ast_visitor.hpp"
+#include <stack>
+
+
+namespace nmodl {
+namespace visitor {
+
+/**
+ * @ingroup visitor_classes
+ * @{
+ */
+
+/**
+ * \brief Visitor that checks the parents
+ */
+class CkParentVisitor : public AstVisitor {
+    private:
+        /**
+        * \brief Keeps track of the parents while going down the tree
+        */
+        ast::Ast* parent = nullptr;
+        bool ckRootParentNull = false;
+    public:
+
+        CkParentVisitor(const bool ckRootParentNull = false) {}
+
+        int lookup(ast::Ast* node);
```

Comment:
yes, the idea is that the visitor throws a runtime error if the parents are not set up correctly. Otherwise, everything is fine. What should I return exactly in this case?